website/blog/2024-02-29-cortex-slack.md

```diff
@@ -0,0 +1,79 @@
+
+We already pull Slack data into Snowflake for basic analytics, but having triage agent that could keep a watchful eye over the Slack, and let us know about things we'd otherwise have missed would help the Developer Experience team do a better job of keeping our finger on the pulse of dbt developers' needs.
+
+Once it was finished, it looked like this:
+
+[ screenshot tk].
+
+Up to once a day, we'll get a post in our internal Slack with links to a handful of interesting threads for each person's focus areas and a brief summary of the discussion so far. From there, we can go deeper by diving into the thread ourselves, wherever it happens to take place. While developing this I found multiple threads that I wouldn't have found any other way (which was itself a problem, since my model filters out threads once a dbt Labs employee is participating in it, so I kept losing all my testing data).
+
+You probably don't have the exact same use case as I do, but you can imagine a wide set of use case for LLM powered analytics engineering:
+
+- A SaaS company could pull information from sales calls or support tickets to gain insight into conversations
+- A mobile app developer might pull in app store reviews for sentiment analysis
+- By calculating the vector embeddings for text, deduplicating similar but nonidentical text becomes more tractable.
+
+## Tips for building LLM-powered dbt models
+
+- **Always build incrementally.** Anyone who's interacted with any LLM-powered tool knows that it can take some time to get results back from a request, and that the results can vary from one invocation to another. For speed, cost and consistency reasons, I implemented both models incrementally even though in terms of row count the tables are tiny. I also added the full-refresh: false config to protect against full refreshes we run to capture late-arriving facts.
+- **Beware of token limits.** Requests that contain [too many tokens](https://docs.snowflake.com/LIMITEDACCESS/cortex-functions#model-restrictions) are truncated, which can lead to unexpected results if the cutoff point is halfway through a message. In future I would first try to use the llama-70b model (~4k token limit), and for unsuccessful rows make a second pass using the mistral-7b model (~32k token limit). Like many aspects of LLM powered workflows, we expect token length constraints to increase substantially in the near term.
+- **Orchestrate defensively, for now**. Because of the above considerations, I've got these steps running in their own dbt Cloud job, triggered by the successful completion of our main project job. I don't want the data team to be freaked out by a failing production run due to my experiments. We use YAML selectors to define what gets run in our default job; I added another selector for these models and then added that selector to the default job's exclusion list. Once this becomes more stable, I'll fold it into our normal job.
```

Comment:
I assume you're using job chaining? Can we say that?

Reply:
No chaining at the moment! I'm only running it once a day so there's not much need

Reply:
If there was other standard transformations that needed to be run on top of the LLM-powered stuff, then we would want that to be happening inside the normal context I think. I'm just keeping these separate to avoid triggering incidents; once I'm confident that isn't happening, there's no real reason to exclude them.

Reply:
oh interesting